### PR TITLE
SeanC/APPEALS-18736 Navigation for Veterans Health Administration Decision Review Queue Link

### PR DESCRIPTION
**Pull request by seancva**
Resolves [APPEALS-18736](https://vajira.max.gov/browse/APPEALS-18736)
And is liked to [this PR in the frontend-toolkit repo](https://github.com/department-of-veterans-affairs/caseflow-frontend-toolkit/pull/102)

# Description
Added a link to the switch product navigation drop down menu for VHA decision review queue

## Acceptance Criteria
- [X] Code compiles correctly
- [X] When user selects the Veterans Health Administration Decision Review Queue menu option from the drop-down menu navigation, they will be directed to the VHA Decision Review Queue page.
- **If the user selects ‘VHA Decision Review Queue’ menu option:**
  - [X] While currently on an Individual Decision Review page, they will be returned to ‘Reviews needing action – In Progress’ page.
  - [X] While on the main VHA Decision Review Queue landing page, the page will refresh the ‘Reviews needing action – In Progress’ page.
- [X] If a user selects a menu navigation option, there will be no modals displayed if a user has entered data to the screen unless they select a ‘Cancel’ link related to their current process.
- [X] The DR queue link should only appear for VHA users who are members of the VHA business line.

## Testing Plan
1. run `make client-install`
2. Login as ACBAUERVVHAH, or any user that is a member of the vha business line
3. Click the "Your Queue" link
4. Click on the "Switch product" button
5. Click on the "VHA Decision Review Queue" link

# Frontend
## User Facing Changes
 - [X] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---
18736_Before|18736_After

## Storybook Story
*For Frontend (Presentationa) Components*
* [X] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)

# Best practices
## Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [X] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [X] No new code climate issues added

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -123,9 +124,19 @@ def application_urls
   helper_method :application_urls
 
   def manage_urls_for_vha(urls)
+    urls << decision_reviews_vha
     urls << intake_application_url if current_user.intake_user?
   end
 
+  def decision_reviews_vha
```
Comment (TylerBroyles) [inline app/controllers/application_controller.rb:131]:
```suggestion
  def decision_reviews_vha_url
```

File: spec/feature/switch_apps_spec.rb [hunk 2]
```diff
@@ -42,13 +68,29 @@
...
+      before do
+        find("a", text: "Switch product").click
+      end
+
+      scenario "Intake" do
+        expect(page).to have_link(vha_user_links[0][:title], href: vha_user_links[0][:link], exact: true)
+      end
+
+      scenario "Decision Reviews Queue" do
+        expect(page).to have_link(vha_user_links[1][:title], href: vha_user_links[1][:link], exact: true)
```
Comment (TylerBroyles) [inline spec/feature/switch_apps_spec.rb:88]: I think there should be at least one negative test. A user that is not VHA should not see the decision review link. This type of test could be added for intake as well.

**Review by TylerBroyles — CHANGES_REQUESTED** [review]
Good job. I left a couple of comments. The testing plan also needs to include the make client-install command.

File: spec/feature/switch_apps_spec.rb [hunk 3]
```diff
@@ -19,15 +20,48 @@
...
       visit "/queue"
+    end
 
-      expect(page).to have_content("Queue")
-
-      find("a", text: "Switch product").click
-      find("a", text: "Caseflow Hearings").click
+    scenario "sees switch product dropdown" do
+      expect(page).to have_link("Switch product", href: "#Switch product", exact: true)
+    end
```
Comment (craigrva) [inline spec/feature/switch_apps_spec.rb:29]: We should roll as many of these scenarios as possible into a single test run to reduce how long it takes to run the tests. Keeping all of the expectations is fine and we can add a comment with each one describing what the scenario currently says.

File: spec/feature/switch_apps_spec.rb [hunk 4]
```diff
@@ -42,13 +76,29 @@
...
     end
 
-    scenario "currently sees switch product dropdown with only queue and and intake" do
-      visit "/decision_reviews/#{vha_business_line.url}"
-      expect(page).to have_current_path("/decision_reviews/#{vha_business_line.url}", ignore_query: true)
-      find("a", text: "Switch product").click
-      check_for_links
+    scenario "see's the switch product dropdown menu" do
+      expect(page).to have_link("Switch product", href: "#Switch product", exact: true)
+    end
```
Comment (craigrva) [inline spec/feature/switch_apps_spec.rb:84]: Same as the comment above, roll as many of these scenarios into a single test as possible

**Review by craigrva — CHANGES_REQUESTED** [review]

**Review by craigrva — APPROVED** [review]

**Pull request merged**